tests/integration/testHelpers.ts

```diff
@@ -330,13 +354,25 @@ export const depositToken = async (
     'getERC20L1Address/getERC20L2Address failed with proper token address'
   )
 
-  const testWalletL2Balance = await l2Token.balanceOf(
-    await l2Signer.getAddress()
+  const tokenBalL2After = await l2Token.balanceOf(
+    destinationAddress || senderAddress
   )
-  expect(
-    testWalletL2Balance.eq(depositAmount),
-    'l2 wallet not updated after deposit'
-  ).to.be.true
+
+  // only check for standard deposits
+  if (!destinationAddress && !ethDepositAmount) {
+    expect(
+      tokenBalL2After.eq(depositAmount),
+      'l2 wallet not updated after deposit'
+    ).to.be.true
+  }
+
+  // batched token+eth
+  if (ethDepositAmount) {
+    expect(
+      ethBalL2After.gte(ethBalL2Before.add(ethDepositAmount)),
```

Comment:
I don't know if we can make it specific to the point where we can do `ethBalL2After.eq(ethBalL2Before.add(ethDepositAmount))` instead of `gte`.

I tried adding leftover gas and other gas combinations but I couldn't get it precisely correct. That's why I used `gte` here.